File: Shortest XY distance in Grid[20-Mar-2023]/java/BruteForce.java


// User function Template for Java

class Solution {
    static class Pair{
        Integer first;
        Integer second;
        
        Pair(Integer x,Integer y)
        {
            first=x;
            second=y;
        }
    }
    static int shortestXYDist(ArrayList<ArrayList<Character>> grid, int N,int M) {
        
         List<Pair> x_indices = new ArrayList<>();
         List<Pair> y_indices = new ArrayList<>();

        for (int i = 0; i < N; i++) {
            for (int j = 0; j < M; j++) {
                if (grid.get(i).get(j) == 'X') {
                    x_indices.add(new Pair(i, j));
                } else if (grid.get(i).get(j) == 'Y') {
                    y_indices.add(new Pair(i, j));
                }
            }
        }

        int ans = Integer.MAX_VALUE;

        for (Pair X : x_indices) {
            int Xi = X.first;
            int Xj = X.second;

            for (Pair Y : y_indices) {
                int Yi = Y.first;
                int Yj = Y.second;

                ans = Math.min(ans, Math.abs(Xi - Yi) + Math.abs(Xj - Yj));
            }
        }

        return ans;
        
    }
};

File: Shortest XY distance in Grid[20-Mar-2023]/c++/BruteForce.cpp
// User function Template for C++

class Solution {
  public:
    int shortestXYDist(vector<vector<char>> grid, int n, int m) {
        vector<pair<int,int>> x_indices;
        vector<pair<int,int>> y_indices;
        
        for(int i=0;i<n;i++)
        {
            for(int j=0;j<m;j++)
            {
                if(grid[i][j]=='X')
                {
                    x_indices.push_back({i,j});
                }
                else if(grid[i][j]=='Y')
                {
                    y_indices.push_back({i,j});    
                }
            }
        }
        
        int ans=INT_MAX;
        
        for(auto & X:x_indices)
        {
            int Xi=X.first;
            int Xj=X.second;
            
            for(auto & Y:y_indices)
            {
                int Yi=Y.first;
                int Yj=Y.second;
                
                ans=min(ans,abs(Xi-Yi)+abs(Xj-Yj));
            }
        }
        
            return ans;
    }
};